### PR TITLE
Add BlindBlockUtil for EIP-4844

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description
Part of #6440:
BlindBlockUtilEip4844
 - SignedBeaconBlockBlinderEip4844
 -  SignedBeaconBlockUnblinderEip4844

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/eip4844/SignedBeaconBlockBlinderEip4844.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import tech.pegasys.teku.spec.schemas.SchemaDefinitionsEip4844;
+
+public class SignedBeaconBlockBlinderEip4844 extends AbstractSignedBeaconBlockBlinder {
+
+  public SignedBeaconBlockBlinderEip4844(final SchemaDefinitionsEip4844 schemaDefinitions) {
+    super(schemaDefinitions);
+  }
+
+  @Override
+  public SignedBeaconBlock blind(final SignedBeaconBlock signedUnblindedBock) {
```
Comment (ajsutton) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/eip4844/SignedBeaconBlockBlinderEip4844.java]: Any reason we can't just use `SignedBeaconBlock.blind` method here?

Comment (zilm13) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/eip4844/SignedBeaconBlockBlinderEip4844.java]: Nice question. Will check why it's not called also in Bellatrix, I've followed

**Comment by zilm13** [comment]
As I removed most blind duplicated code I'm not sure we still need 3 classes there, but at least it checks that correct Schema definition is passed in

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/BlindBlockUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            blindSignedBeaconBlock,
+            unblinder -> {
+              final BeaconBlock block = unblinder.getSignedBlindedBeaconBlock().getMessage();
+
+              if (block
+                  .getBody()
+                  .getOptionalExecutionPayloadHeader()
+                  .orElseThrow()
+                  .isHeaderOfDefaultPayload()) {
+                // Terminal block not reached, provide default payload
```
Comment (tbenr) [inline ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/BlindBlockUtilTest.java]: youre providing a random block, so you'll never fall in this code path so you can leave only the code in the `else`

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/BlindBlockUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  void setUp(final SpecContext specContext) {
+    spec = specContext.getSpec();
+    dataStructureUtil = specContext.getDataStructureUtil();
+    final SpecVersion specVersion = spec.forMilestone(specContext.getSpecMilestone());
+    blindBlockUtil = specVersion.getBlindBlockUtil().orElseThrow();
+  }
+
+  @TestTemplate
+  void shouldBlindAndUnblindBlock(final SpecContext specContext) {
+    final SignedBeaconBlock signedBeaconBlock = dataStructureUtil.randomSignedBeaconBlock();
```
Comment (tbenr) [inline ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/BlindBlockUtilTest.java:46]: CodeQL is actually right :)

**Review by tbenr — APPROVED** [review]
LGTM, just two nits

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/BlindBlockUtilTest.java (hunk 3 above)
Comment (zilm13) [reply ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/BlindBlockUtilTest.java:46]: fixed

**Pull request merged**